src/App.tsx

```diff
@@ -1,29 +1,54 @@
-import React from 'react';
+import React, { useState } from 'react';
 import './App.css';
 import { getNumbers } from './utils';
+import { Pagination } from './components/Pagination';
 
 // eslint-disable-next-line @typescript-eslint/no-unused-vars
-const items = getNumbers(1, 42).map(n => `Item ${n}`);
 
 export const App: React.FC = () => {
+  const [currentPage, setCurrentPage] = useState(1);
+  const [perPage, setPerPage] = useState(5);
+  const items = getNumbers(1, 42).map(n => `Item ${n}`);
+  const total = items.length;
+  const itemsPerPageOptions = [3, 5, 10, 20];
+
+  const handlePageChange = (page: number) => {
+    setCurrentPage(page);
+  };
+
+  const handlePerPageChange = (newPerPage: number) => {
+    setPerPage(newPerPage);
+    setCurrentPage(1);
+  };
+
+  const startIndex = (currentPage - 1) * perPage + 1;
+  const endIndex = Math.min(startIndex + perPage - 1, total);
+  const currentItems = items.slice(startIndex - 1, endIndex);
```

Comment:
The calculation for `endIndex` does not account for the case where `perPage` is greater than the total number of items. While `Math.min` is used to ensure `endIndex` does not exceed `total`, there is no similar check for `startIndex`. If `currentPage` is set to a value that causes `startIndex` to be greater than `total`, `items.slice` will return an empty array, which may not be the intended behavior if there are items to display. Consider adding a check to ensure `startIndex` is within the bounds of the items array.